### PR TITLE
Remove uses of deprecated JSON API

### PR DESCRIPTION
**Pull request by marcomorain**
The version 3.12 of the Java driver deprecates `com.mongodb.util.JSON`
in favour of the JSON functions on the DBObject class. The deprecated
API will be removed in driver version 4.x.

In order to prepare for the upgrade to 4.x we can upgrade from driver
`3.10` to `3.12` and remove calls to the decprecated API.

### CODE REVIEW

**Comment by marcomorain** [comment]
@circlecai this is my first shot at the upgrade. We should be able to pair on this this week.

File: project.clj [hunk 1]
```diff
@@ -10,7 +10,7 @@
             :distribution :repo}
   :min-lein-version "2.0.0"
   :dependencies [[org.clojure/data.json "0.2.7"]
```
Comment (jheander) [inline project.clj:12]: Maybe also update the org.clojure/data.json to a more recent version?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+version: '2'
+services:
+  mongodb:
+    image: mongo:3.6
```
Comment (jheander) [inline docker-compose.yml:4]: What is the docker-compose file used for?

Comment (circlecai) [reply docker-compose.yml:4]: To run tests locally. We can take it out of this PR if you think it's not valuable, or if it should be introduced separately with documentation.

**Review by jheander — APPROVED** [review]

**Pull request merged**